### PR TITLE
Step 16: User functionality

### PR DESCRIPTION
**Pull request by taheryasir**
- [x] Let's create a user model
- [x] Let's create the first user with seed
- [x] Let's connect users and tasks
- [x] Index for associations

task link: https://github.com/Fablic/training/blob/develop_en/README.md#step-16-make-it-available-to-multiple-people-introduction-of-user-feature

### CODE REVIEW

**Review by shunshunsuke — COMMENTED** [review]
Good job!
please check it out.

File: myapp/db/migrate/20210625050724_add_user_id_to_tasks.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserIdToTasks < ActiveRecord::Migration[6.0]
+  def change
+    add_column :tasks, :user_id, :integer
```
[the remark this answers is not in the document]
Comment (taheryasir) [reply myapp/db/migrate/20210625050724_add_user_id_to_tasks.rb:3]: I did that's why this file was created (or i am missing something )

File: myapp/db/migrate/20210625120923_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :users do |t|
```
[the remark this answers is not in the document]
Comment (taheryasir) [reply myapp/db/migrate/20210625120923_create_users.rb:3]: added it on model

File: myapp/db/migrate/20210625050724_add_user_id_to_tasks.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (shunshunsuke) [reply myapp/db/migrate/20210625050724_add_user_id_to_tasks.rb:3]: @taheryasir 
There is not an index on user_id.



I would recommend using '[add_reference](https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_reference)'.
like this
```
add_reference :tasks, :user, foreign_key: true
```

**Review by shunshunsuke — APPROVED** [review]
LGTM!
LGTM

**Pull request merged**